opera/rules_test.go

```diff
@@ -0,0 +1,184 @@
+package opera
+
+import (
+	"reflect"
+	"testing"
+)
+
+func TestRules_Copy_DeepCopied(t *testing.T) {
```

Comment:
nit: we try to use full sentences in the test name, to describe the property tested:
ideas: 
```suggestion
func TestRules_Copy_CopiesAreDisjoint(t *testing.T) {
```
```suggestion
func TestRules_Copy_CopiesShareNoData(t *testing.T) {
```

```suggestion
func TestRules_Copy_IsImplementedAsDeepCopy(t *testing.T) {
```